app/models/service_template_ansible_playbook.rb

```diff
@@ -179,6 +180,11 @@ def retirement_potential?
 
   private
 
+  def retain_resource_actions
+    @resource_actions = resource_actions
+    yield
+  end
```

Comment:
How about

```
@resource_actions = resource_actions
yield
delete_job_templates
```

Easier to read, logic is pulled out of the callback.
Reuses a tested method.